napari/components/layerlist.py

```diff
@@ -310,7 +310,7 @@ def extent(self) -> Extent:
         return self.get_extent(list(self))
 
     @property
-    def _ranges(self) -> List[Tuple[float, float, float]]:
+    def _ranges(self) -> Tuple[Tuple[float, float, float]]:
```

Comment:
Do we know the size of the external tuple, and thus can this be made:

```
T3 = Tuple[float, float, float]
def _ranges(self) -> Tuple[T3, T3, T3]:
```
?

Reply:
Re-reading the code I think I'm wrong, but in which case it should maybe be 
```suggestion
    def _ranges(self) -> Tuple[Tuple[float, float, float],...]:
```

To indicate it's variable length ?

Reply:
Yes, I think you're right it has to be variable because self.extent can be 2D or 3D.